postgres/codenamemap.yaml

```diff
@@ -14,12 +17,13 @@
   #}
 
   {# use upstream version if configured #}
-  {% set version = upstream_version|default(version) %}
+  {% if use_upstream_repo %}
+    {% set version = upstream_version %}
+  {% endif %}
 
 {{ codename|default(name, true) }}:
   # PostgreSQL packages are mostly downloaded from `main` repo component
-  pkg_repo: 'deb http://apt.postgresql.org/pub/repos/apt/ {{ name }}-pgdg main {{ version }}'
-  pkg_repo_humanname: PostgreSQL Official Repository
+  pkg_repo: 'deb http://apt.postgresql.org/pub/repos/apt {{ name }}-pgdg main {{ upstream_version }}'
```

Comment:
Because `pkg_repo` parameter is only applicable for upstream repository, there is no point to set it's component name to the version provided by OS distribution.
But the main point here is that this variable also playing its part in *removing* upstream repository configuration when `use_upstream_repo` set to `False`.

Basically, before installing any distro supplied packages, `upstream.sls` will make sure that upstream `deb` source record and GPG key are removed. Because package names from upstream could possibly match the ones from the "stock", it's really important to avoid installation of newer tools or libs which would be incompatible with PostgreSQL server installed from distribution archive.